src/shogun/kernel/PeriodicKernel.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2015 Esben Soerig
+ */
+
+#include <shogun/lib/common.h>
+#include <shogun/base/Parameter.h>
+#include <shogun/features/DotFeatures.h>
+#include <shogun/io/SGIO.h>
+#include "PeriodicKernel.h"
+
+using namespace shogun;
+
+CPeriodicKernel::CPeriodicKernel() : CDotKernel()
+{
+	init();
+}
+
+CPeriodicKernel::CPeriodicKernel(int32_t size, float64_t w, float64_t p) : CDotKernel(size)
+{
+	init();
+	set_width(w);
+	set_period(p);
+}
+
+CPeriodicKernel::CPeriodicKernel(CDotFeatures* l, CDotFeatures* r,
+		float64_t w, float64_t p, int32_t size) : CDotKernel(size)
+{
+	init();
+	set_width(w);
+	set_period(p);
+	init(l,r);
+}
+
+CPeriodicKernel::~CPeriodicKernel()
+{
+	cleanup();
+}
+
+CPeriodicKernel* CPeriodicKernel::obtain_from_generic(CKernel* kernel)
+{
+	// TODO: Add proper periodic kernel enum value
+	if (kernel->get_kernel_type()!= K_PERIODIC)
+	{
+		SG_SERROR("CPeriodicKernel::obtain_from_generic(): provided kernel is "
+				"not of type CPeriodicKernel!\n");
+	}
+
+	/* since an additional reference is returned */
+	SG_REF(kernel);
+	return (CPeriodicKernel*)kernel;
+}
+
+void CPeriodicKernel::cleanup()
+{
+	if (sq_lhs != sq_rhs)
+		SG_FREE(sq_rhs);
+	sq_rhs = NULL;
+
+	SG_FREE(sq_lhs);
+	sq_lhs = NULL;
+
+	CKernel::cleanup();
+}
+
+void CPeriodicKernel::precompute_squared_helper(float64_t* &buf, CDotFeatures* df)
+{
+	ASSERT(df)
```

Comment:
We try not to use ASSERT anymore, see here how to write error messages:
https://github.com/shogun-toolbox/shogun/wiki/Assertions